src/modules/RootPart.rb

```diff
@@ -1675,27 +1677,53 @@ def MountPartitions(root_device_current)
         end
         Update.clean_backup
         create_backup
+        inject_intsys_files
       end
 
       success
     end
 
+    RESOLV_CONF = "/etc/resolv.conf".freeze
+
     # known configuration files that are changed during update, so we need to
     # backup them to restore if something goes wrong (bnc#882039)
     BACKUP_DIRS = {
-      "sw_mgmt" => [
+      # use a number prefix to set the execution order
+      "0100-sw_mgmt" => [
         "/var/lib/rpm",
         "/etc/zypp/repos.d",
         "/etc/zypp/services.d",
         "/etc/zypp/credentials.d"
+      ],
+      # this should be restored as the very last one, after restoring the original
+      # resolv.conf the network might not work properly in the chroot
+      "0999-resolv_conf" => [
+        RESOLV_CONF
       ]
     }
+
+    # copy these device files into inst-sys to make sure the chrooted scripts work
+    REQUIRED_DEVICES = [ "null", "zero", "random", "urandom" ]
+
     def create_backup
       BACKUP_DIRS.each_pair do |name, paths|
         Update.create_backup(name, paths)
       end
     end
 
+    # inject the required files from the inst-sys to the chroot so
+    # the network connection works for the chrooted scripts
+    def inject_intsys_files
+      # the original file is backed up and restored later
+      ::FileUtils.cp(RESOLV_CONF, File.join(Installation.destdir, RESOLV_CONF)) if File.exist?(RESOLV_CONF)
+
+      REQUIRED_DEVICES.each do |dev|
+        # FileUtils.cp does not handle device files correctly, use `cp -a` instead
+        target = File.join(Installation.destdir, "dev", dev)
+        system("cp -a /dev/#{dev} #{Shellwords.escape(target)}")
```

Comment:
I really do not like it, as it is not much robust and contain hardcoded list of devices. I usually use much simplier `mount -o bind /dev/ /mnt/dev`

Reply:
Hm, it's a bit tricky. How does the `mount --bind` work with `rm -rf /mnt/dev` [here](https://github.com/yast/yast-packager/blob/03dd0f404ac08ae1e11bdd266e6d7db33d9d4034/src/clients/inst_kickoff.rb#L54-L55)?

Originally I used `mount -t devtmpfs devtmpfs /mnt/dev` but it turned out that this `rm -rf /mnt/dev` call also deleted the `/dev` in the inst-sys breaking installation...

I'd like to use a better solution, but it's too late for SLE15...